pilot/pkg/networking/core/v1alpha3/envoyfilter/rc_patch.go

```diff
@@ -151,6 +151,8 @@ func patchHTTPRoutes(patchContext networking.EnvoyFilter_PatchContext,
 	routeConfiguration *route.RouteConfiguration, virtualHost *route.VirtualHost, portMap model.GatewayPortMap,
 ) {
 	routesRemoved := false
+	// different virtualHosts may share same routes pointer
+	virtualHost.Routes = cloneVhostRoutes(virtualHost.Routes)
```

Comment:
This is actually more expensive for the cases where we do not patch anything (common case), maybe we can make it a bit more conditional?

Reply:
> This is actually more expensive for the cases where we do not patch anything (common case), maybe we can make it a bit more conditional?

It is true, it will be a big improvement in the scenario where our large number of routers need merge every time, but it will be more expensive when merge is not needed, I will try to add some conditions to him